tests/test_packageAlerts.py

```diff
@@ -275,9 +330,122 @@ def testMakeAlertDict(self):
             self.assertEqual(alert["cutoutTemplate"],
                              cutoutBytes)
 
-    def testRun(self):
-        """Test the run method of package alerts.
+    def test_produceAlerts_empty_password(self):
+        """ Test that produceAlerts raises if the password is empty or missing.
+        """
+        self.environ['AP_KAFKA_PRODUCER_PASSWORD'] = ""
+        with self.assertRaisesRegex(ValueError, "Kafka password"):
+            packConfig = PackageAlertsConfig(doProduceAlerts=True)
+            packageAlerts = PackageAlertsTask(config=packConfig)  # noqa: F841
+
+        del self.environ['AP_KAFKA_PRODUCER_PASSWORD']
+        with self.assertRaisesRegex(ValueError, "Kafka password"):
+            packConfig = PackageAlertsConfig(doProduceAlerts=True)
+            packageAlerts = PackageAlertsTask(config=packConfig)  # noqa: F841
+
+    def test_produceAlerts_empty_username(self):
+        """ Test that produceAlerts raises if the username is empty or missing.
+        """
+        self.environ['AP_KAFKA_PRODUCER_USERNAME'] = ""
+        with self.assertRaisesRegex(ValueError, "Kafka username"):
+            packConfig = PackageAlertsConfig(doProduceAlerts=True)
+            packageAlerts = PackageAlertsTask(config=packConfig)  # noqa: F841
```

Comment:
Don't assign to `packageAlerts` here: the raised exception means that the task instantiation doesn't return anything. Then you don't have to have noqa for all of these.

Reply:
Fixed